### DISCUSSION TITLE
Do not use hardcoded paths

### DISCUSSION THREAD

**Discussion by Friend-LGA** [quasarframework/quasar#17517] [Ideas / Proposals]
In the file `@quasar/app-vite/lib/app-paths.js` is a list of hardcoded paths:
```js
const cliDir = resolve(__dirname, '..')
const publicDir = resolve(appDir, 'public')
const srcDir = resolve(appDir, 'src')
const pwaDir = resolve(appDir, 'src-pwa')
const ssrDir = resolve(appDir, 'src-ssr')
const cordovaDir = resolve(appDir, 'src-cordova')
const capacitorDir = resolve(appDir, 'src-capacitor')
const electronDir = resolve(appDir, 'src-electron')
const bexDir = resolve(appDir, 'src-bex')
```

I believe that such paths should be configurable using config file `quasar.config.js`.

For example I'm making an Electron app and would like to name root folder `src_system_back` instead of `src-electron` and couldn't find a way. You actually have options to change `main.js` and `preload.js` file paths, but not actually paths, just names, which is misleading and ambiguous:
```js
// quasar.config.js
sourceFiles: {
  electronMain: 'src-electron/main',
  electronPreload: 'src-electron/preload',
},
```
If I will change this `src-electron` path prefix, than quasar will just recreate `src-electron` folder.

Also I am using `snake_case` files name convention and Quasar forces me to break the rule for just this one folder, which is really annoying.

I do believe that frameworks must provide such basic configurations out of the box.

**Comment by dmke** [comment]
Not to hijack this discussion, but I have a similar feature request in #17503 (making `appDir` configurable). I believe any solution should cover both cases.
